### ISSUE TITLE
useMutation fails when used inside another function

### ISSUE THREAD

**Issue by TheDutchCoder** [Akryum/vue-apollo#1319]
**Describe the bug**
When using `useMutation` in a `<script setup>`, the function only works when directly called at the root level, but not when called from another function.

**To Reproduce**
```js
// This works
const { mutate: updateProfile } = useMutation(gql`
  mutation updateProfile ($accountId: String!, $name: String) {
    updateProfile (accountId: $accountId, name: $name) {
      accountId,
    }
  }
`)

updateProfile({
  accountId: '123',
  bannerLoc: 'Test',
})

// This does not
const foo = () => {
  const { mutate: updateProfile } = useMutation(gql`
    mutation updateProfile ($accountId: String!, $name: String) {
      updateProfile (accountId: $accountId, name: $name) {
        accountId,
      }
    }
  `)

  updateProfile({
    accountId: '123',
    bannerLoc: 'Test',
  })
}

foo()
```

**Expected behavior**
For this to work.

**Versions**
vue: 3.2.20
@vue/apollo-composable: 4.0.0-alpha-16
@apollo/client: 3.5.8

**Comment by TheDutchCoder** [comment]
The same happens when using regular `defineComponent` with a `setup` method.
`useMutation` just can't get called from another function, which doesn't seem to make sense.

**Comment by Akryum** [comment]
Please provide a runnable reproduction. Thanks

**Comment by TheDutchCoder** [comment]
Ok I think I found the issue:
the mutation has to be declared in the setup scope, but the function can be called in another function:
```js
const { mutate: updateProfile } = useMutation(gql`
  mutation updateProfile ($accountId: String!, $name: String) {
    updateProfile (accountId: $accountId, name: $name) {
      accountId,
    }
  }
`)

const upload = () => {
  updateProfile({
    accountId: '123',
    name: 'Test',
  })
}
```

Maybe this can be clarified in the docs?

**Comment by TheDutchCoder** [comment]
I'll close it as it's not directly an issue, but some clarification in the docs would be good. If I have some time, I can PR something!

**Comment by VincentLandais** [comment]
I am currently facing this issue. In my situation, I can't call `useMutation` outside the function. The error said ```Uncaught (in promise) Error: Apollo client with id default not found. Use provideApolloClient() if you are outside of a component setup.```
So I added  `provideApolloClient(DefaultApolloClient)` at the top of the function. Now the call to `useMutation` works fine but I can't call the returned `mutate` function.

This is the error i'm getting.

<img width="672" alt="Screenshot 2022-05-13 at 10 37 24" src="https://user-images.githubusercontent.com/100678450/168245241-8868eb3c-6ec6-4b24-8124-3b8d8a5c31c3.png">

I checked the type of `mutate` and it's definitely a function.
Line 140 from my `HomeView` component is the call to `mutate`
